st2common/tests/unit/test_operators.py

```diff
@@ -19,16 +19,540 @@
 from st2common.util import date as date_utils
 
 
+def dict_strict_equal(d1, d2):
+    """
+    Compare two dictionaries, ensuring that all keys exist in both and that the
+    values for each key are the same.
+    """
+    if len(d1.keys()) != len(d2.keys()):
```

Comment:
I believe `dict1 == dict2` should work in the same manner - aka deep strict comparison should be performed by default.

Would be good to confirm that's indeed the case because I would imagine if that's the case, Python implementations would optimize that use-case and make it fast and efficient :)

Reply:
You are correct - Python's `==` for dicts checks the key and values for equality, not just the `id()` or the `hash()` as I initially thought.

I'll fix this in another PR (should simply be able to remove the entire `dict_strict_equal` function, its test case, and convert `list_of_dicts_strict_equal` to use `==` instead).

Reply:
PR #3839 fixes this.